.changeset/afraid-singers-arrive.md

```diff
@@ -0,0 +1,5 @@
+---
+"@apollo/query-planner": patch
+---
+
+Fix bug in query planning when a subgraph jump for @requires can sometimes try to fetch @key fields from a subgraph that doesn't have them.
```

Comment:
Why don't we mention the error message here? It's a clearly recognizable symptom, even though this PR may not fix every one of them.

Reply:
That's fair, I've updated the changeset to include an example error message.